src/tests/ftest/util/slurm_utils.py

```diff
@@ -1,43 +1,94 @@
 
-  Unless required by applicable law or agreed to in writing, software
-  distributed under the License is distributed on an "AS IS" BASIS,
-  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-  See the License for the specific language governing permissions and
-  limitations under the License.
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
 
-  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
-  The Government's rights to use, modify, reproduce, release, perform, display,
-  or disclose this software are subject to the terms of the Apache License as
-  provided in Contract No. B609815.
-  Any reproduction of computer software, computer software documentation, or
-  portions thereof marked with this legend must also reproduce the markings.
-'''
+GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+The Government's rights to use, modify, reproduce, release, perform, display,
+or disclose this software are subject to the terms of the Apache License as
+provided in Contract No. B609815.
+Any reproduction of computer software, computer software documentation, or
+portions thereof marked with this legend must also reproduce the markings.
+"""
 
 from __future__ import print_function
 import os
 import random
 import time
 import subprocess
 import threading
-import pyslurm
+import pyslurm  # pylint: disable=import-error
+from avocado.utils import process
+
+W_LOCK = threading.Lock()
 
-_lock = threading.Lock()
 
 class SlurmFailed(Exception):
-    """ Thrown when something goes wrong with slurm """
+    """Thrown when something goes wrong with slurm."""
 
-def write_slurm_script(path, name, output, nodecount, cmds, sbatch=None):
+
+def cancel_jobs(job_id):
+    """Cancel slurms jobs.
+
+    :param job_id: slurm job id
+    :type job_id: int
+    :return: status
+    :rtype: bool
+
+    """
+    status = process.system("scancel {}".format(job_id))
+    if status > 0:
+        raise SlurmFailed(
+            "Slurm: scancel failed to kill job {}".format(job_id))
+    return status
+
+
+def create_slurm_partition(nodelist, name):
+    """Create a slurm partion for soak jobs.
+
+    client nodes will be allocated for this partiton
+    Args:
+        nodelist (list): list of nodes for job allocation
+        name (str): partition name
+    Returns: status bool
+    """
+    # If the partition exists; delete it because if may have wrong nodes
+    status = process.system(
+        "scontrol delete PartitionName={}".format(name))
+    if status == 0:
+        status = process.system(
+            "scontrol create PartitionName={} Nodes={}".format(
+                name, ",".join(nodelist)))
+    return status
+
+
+def delete_slurm_partition(name):
```

Comment:
Probably not appropriate --partition should exist ahead of time.

Reply:
I will change this since I can not create the partition on the dudley fabric as a user